### PR TITLE
Update covid-xray example to use dash-slicer

### PR DESCRIPTION
**Pull request by almarklein**
Issue for app: #[issue number here]

# App pull request
- [ ] This is a new app
- [x] I am improving an existing app

## About

- Playground deployment URL (new version): ?
- Current gallery app URL: https://dash-gallery.plotly.host/dash-covid-xray/

## Description

This modifies the Covid xray example to use the new [dash-slicer](https://github.com/plotly/dash-slicer), removing quite a bit of boilerplate code, so the example focus becomes more at the image processing logic.

Comments:
* The dash-slicer project is not yet on Pypi; the purpose of this rewrite is to tweak the API to accommodate apps like this :)
* This currently relies on features of dash-slider that are not yet in the main branch. In other words, it only really Works On My Machine (TM).
* As long as the above is the case, this stays a Draft PR.

### CODE REVIEW

File: apps/dash-covid-xray/app.py [hunk 1]
```diff
@@ -1,132 +1,79 @@
...
+
+def path_to_coords(path):
+    """From SVG path to numpy array of coordinates, each row being a (row, col) point"""
     indices_str = [
         el.replace("M", "").replace("Z", "").split(",") for el in path.split("L")
     ]
     return np.array(indices_str, dtype=float)
 
 
-def make_figure(
```
Comment (emmanuelle) [inline apps/dash-covid-xray/app.py:62]: I'm so glad this big chunk of code can go away :-).

**Comment by emmanuelle** [comment]
Thanks for opening this PR, it shows really well how the code will be simplified with dash-slicer. I've just noticed two little things:
- annotations don't persist on different slices (whereas they do in the deployed app), which is normal because the internal callback updating the figure does not know about these annotations. Since the case of persisting annotations is probably a bit specific I think I'm ok with this change but we can think a little bit if there would be some not-too-hacky way of making some annotations persist (like the overlay mask maybe?)
- I've noticed a small bug that the overlay mask is not visible everywhere it should be, like in this screenshot (I think it should be visible on the left image, according to the indicators position)
image

**Comment by emmanuelle** [comment]
@almarklein I created #538 to work on a branch on `origin`. I propose to take a shot at finishing the PR now that the slicer is released and I'd like to write my blog post about the app in the next days.

**Comment by almarklein** [comment]
Superseded by #538

**Pull request closed without merging**